sympy/solvers/ode/systems.py

```diff
@@ -0,0 +1,375 @@
+from sympy import (Matrix, Derivative, Symbol)
+from sympy.solvers.deutils import ode_order
+from sympy.core.numbers import I
+from sympy.core.relational import Eq
+from sympy.core.symbol import Dummy
+from sympy.functions import exp, im, cos, sin, re
+from sympy.functions.combinatorial.factorials import factorial
+from sympy.matrices import zeros
+from sympy.utilities import numbered_symbols, default_sort_key
+from sympy.utilities.iterables import uniq
+from sympy.simplify import simplify
+
+
+def _get_func_order(eqs, funcs):
+    order = dict()
+    func_dict = dict()
+
+    for func in funcs:
+        if not order.get(func, False):
+            max_order = -1
+            eq_no = -1
+            for i, eq in enumerate(eqs):
+                order_ = ode_order(eq, func)
+                if max_order < order_:
+                    max_order = order_
+                    eq_no = i
+            if eq_no in func_dict:
+                list_func = []
+                list_func.append(func_dict[eq_no])
+                list_func.append(func)
+                func_dict[eq_no] = list_func
+            else:
+                if eq_no >= 0:
+                    func_dict[eq_no] = func
+            if max_order >= 0:
+                order[func] = max_order
+
+    return order
+
+
+def matrix_exp(A, t):
+    '''Matrix exponential exp(A*t) for the matrix A and scalar t.
```

Comment:
This docstring should follow the style guide:
https://docs.sympy.org/latest/documentation-style-guide.html

Also actually this function probably should be added to `ode.rst` in the system section.